src/matrix.rs

```diff
@@ -99,11 +99,29 @@ impl<S> Matrix2<S> {
 }
 
 impl<S: BaseFloat> Matrix2<S> {
-    /// Create a transformation matrix that will cause a vector to point at
-    /// `dir`, using `up` for orientation.
+    /// Create a transformation matrix that will cause `unit_x()` to point at
+    /// `dir`. `unit_y()` will be perpendicular to `dir`, and the closest to `up`.
     pub fn look_at(dir: Vector2<S>, up: Vector2<S>) -> Matrix2<S> {
-        //TODO: verify look_at 2D
-        Matrix2::from_cols(up, dir).transpose()
+        let basis1 = dir.normalize();
+        let basis2 = if up.x * basis1.y >= up.y * basis1.x {
```

Comment:
Sounds good, it's way better to call into the same function for the common part. I think the only concern with that was that it would call `noramlize` on an already normalized vector, we would first normalize in `look_at` and then again when it calls `look_at_stable`.

Would you prefer if I
Option 1:  made a private function that is like `look_at_stable`, but assumes that the vector is normalized, that each of these call into, or
Option 2: ignored the duplicate work, and call `look_at_stable` from `look_at` as you describe.